vars/signArtifacts.groovy

```diff
@@ -0,0 +1,25 @@
+void call(Map args = [:]) {
+
+    environment {
+        // These ENV variables are required by https://github.com/opensearch-project/opensearch-signer-client
+        // This client is invoked internally by the sign script.
+        ROLE = "${SIGNER_CLIENT_ROLE}"
+        EXTERNAL_ID = "${SIGNER_CLIENT_EXTERNAL_ID}"
+        UNSIGNED_BUCKET = "${SIGNER_CLIENT_UNSIGNED_BUCKET}"
+        SIGNED_BUCKET = "${SIGNER_CLIENT_SIGNED_BUCKET}"
+    }
+
+    if( !fileExists("$WORKSPACE/sign.sh")) {
+        git url: 'https://github.com/opensearch-project/opensearch-build.git', branch: 'main'
+    }
+
+    if( !fileExists("$WORKSPACE/opensearch.pgp")) {
```

Comment:
Does it hurt to reimport? We probably need to do it anyway if the key were to change.

Reply:
Same reason, we would want to use the same key for one running instance of a job

Reply:
Same as above.

Reply:
This should be done once in the job, not every time we try to sign something. Same as above, neither will solve runs across stages.

Reply:
Again, we want to make this function self sufficient and not rely on the calling functions for a setup